lib/unlock-helpers.js

```diff
@@ -4,27 +4,41 @@ import _ from 'lodash';
 import { util } from 'appium-support';
 
 const PIN_UNLOCK = 'pin';
+const PIN_UNLOCK_KEY_EVENT = 'pinWithKeyEvent';
 const PASSWORD_UNLOCK = 'password';
 const PATTERN_UNLOCK = 'pattern';
 const FINGERPRINT_UNLOCK = 'fingerprint';
-const UNLOCK_TYPES = [PIN_UNLOCK, PASSWORD_UNLOCK, PATTERN_UNLOCK, FINGERPRINT_UNLOCK];
+const UNLOCK_TYPES = [PIN_UNLOCK, PIN_UNLOCK_KEY_EVENT, PASSWORD_UNLOCK, PATTERN_UNLOCK, FINGERPRINT_UNLOCK];
 const KEYCODE_NUMPAD_ENTER = 66;
 const KEYCODE_POWER = 26;
 const KEYCODE_WAKEUP = 224; // Can work over API Level 20
 const UNLOCK_WAIT_TIME = 100;
 const HIDE_KEYBOARD_WAIT_TIME = 100;
 const INPUT_KEYS_WAIT_TIME = 100;
+// mapping of Android keycode and numbers
+const ANDROID_KEY_CODE_NUMBER = {
+  '0': 7,
```

Comment:
I believe this mapping could be generated dynamically, there is no need to hardcode each item

Reply:
```
const ZERO_KEY_CODE = 7;
result = _.range(10).reduce((acc) => {
  acc[String.fromCharCode('0'.charCodeAt(0) + _.size(acc))] = ZERO_KEY_CODE + _.size(acc);
  return acc;
}, {});
```